lib/IRGen/GenReflection.cpp

```diff
@@ -906,7 +906,6 @@ emitAssociatedTypeMetadataRecord(const ProtocolConformance *Conformance) {
 void IRGenModule::emitBuiltinReflectionMetadata() {
   if (getSwiftModule()->isStdlibModule()) {
     BuiltinTypes.insert(Context.TheNativeObjectType);
-    BuiltinTypes.insert(Context.TheUnknownObjectType);
```

Comment:
Can you try something like BuiltinTypes.insert(Context.getAnyObjectType()->getCanonicalType())?

Reply:
... it would have to emit the descriptor under the old name "BO" too. Also see my comments in Reflection/TypeLowering.cpp